File: app/src/main/java/android/arch/paging/ShiftedListUpdateCallback.java
package android.arch.paging;


import android.support.v7.util.ListUpdateCallback;
import android.support.v7.widget.RecyclerView;

public class ShiftedListUpdateCallback implements ListUpdateCallback {
    private final RecyclerView.Adapter mAdapter;

    public ShiftedListUpdateCallback(RecyclerView.Adapter adapter) {
        mAdapter = adapter;
    }

    @Override
    public void onInserted(int position, int count) {
        mAdapter.notifyItemRangeInserted(position+1, count);
    }

    @Override
    public void onRemoved(int position, int count) {
        mAdapter.notifyItemRangeRemoved(position + 1, count);
    }

    @Override
    public void onMoved(int fromPosition, int toPosition) {
        mAdapter.notifyItemMoved(fromPosition+1, toPosition);
    }

    @Override
    public void onChanged(int position, int count, Object payload) {
        mAdapter.notifyItemRangeChanged(position+1, count, payload);
    }
}

File: app/src/main/java/com/letitplay/maugry/letitplay/user_flow/ShiftedListUpdateCallback.kt
package com.letitplay.maugry.letitplay.user_flow


import android.support.v7.util.ListUpdateCallback
import android.support.v7.widget.RecyclerView

class ShiftedListUpdateCallback(private val mAdapter: RecyclerView.Adapter<*>) : ListUpdateCallback {

    override fun onInserted(position: Int, count: Int) {
        mAdapter.notifyItemRangeInserted(position + 1, count)
    }

    override fun onRemoved(position: Int, count: Int) {
        mAdapter.notifyItemRangeRemoved(position + 1, count)
    }

    override fun onMoved(fromPosition: Int, toPosition: Int) {
        mAdapter.notifyItemMoved(fromPosition + 1, toPosition)
    }

    override fun onChanged(position: Int, count: Int, payload: Any) {
        mAdapter.notifyItemRangeChanged(position + 1, count, payload)
    }
}
